File: TasSync/beis-updates19Mar21(2)/SP/src/types/NAOUpdateEvidence.ts
import { IEntity } from "./Entity";
import { User } from "./User";

export interface INAOUpdateEvidence  extends IEntity {
    IsLink?:boolean;
    Details?: string;
    AdditionalNotes?: string;
    NAOUpdateId?: number; 
    DateUploaded?: Date;
    UploadedByUserId?: number;
    //User?: User;

}

export class NAOUpdateEvidence implements INAOUpdateEvidence{ 
    public ID: number = 0;
    public Title: string = null;
    public Details?: string = "";
    public AdditionalNotes?: string = "";
    public NAOUpdateId?: number = null;
    public DateUploaded?: Date = null;
    public UploadedByUserId?: number = null;

    constructor(naoUpdateId: number) {
        this.NAOUpdateId = naoUpdateId;
    }

    //public User?: User = null; 
}

File: ControlAssuranceAPI/Models/NAOUpdateEvidence.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class NAOUpdateEvidence
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public bool? IsLink { get; set; }

    public string? Details { get; set; }

    public string? AdditionalNotes { get; set; }

    public int? NAOUpdateId { get; set; }

    public DateTime? DateUploaded { get; set; }

    public int? UploadedByUserId { get; set; }

    public virtual NAOUpdate? NAOUpdate { get; set; }

    public virtual User? User { get; set; }
}
